src/wutil.cpp

```diff
@@ -564,7 +564,7 @@ file_id_t file_id_t::file_id_from_stat(const struct stat *buf)
 #if defined(__APPLE__) || defined(__DragonFly__) || defined(__FreeBSD__) ||  defined(__OpenBSD__) || defined(__NetBSD__)
     result.generation = buf->st_gen;
 #else
-    result.generation = 0;
+    result.generation = NULL;
```

Comment:
The place to look if you really want to see something besides zeros there will be NFS and userspace filesystems.

The standard says that field will be there and it's accessed by standard means so I see no reason not to store the data from the struct even if it might be zero'd.